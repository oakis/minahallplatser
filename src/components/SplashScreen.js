import React, { Component } from 'react';
import { View, ImageBackground, Text, AsyncStorage } from 'react-native';
import { Actions } from 'react-native-router-flux';
import firebase from 'firebase';
import { connect } from 'react-redux';
import { autoLogin } from '../actions';
import { Spinner } from './common';
import { colors } from './style';

class SplashScreen extends Component {

	componentDidMount() {
		// Try to automaticly login
		firebase.auth().onAuthStateChanged((fbUser) => {
			AsyncStorage.getItem('minahallplatser-user')
			.then((dataJson) => {
				const user = JSON.parse(dataJson);
				if (user.uid === fbUser.uid) {
					this.props.autoLogin(fbUser);
				} else {
					Actions.auth();
				}
			})
			.catch(() => {
				Actions.auth();
			});
		}, (err) => window.log(err));
	}

	render() {
		return (
			<ImageBackground
				source={{ uri: 'https://www.w3schools.com/css/img_fjords.jpg' }}
				style={{
					flex: 1
				}}
			>
				<View
					style={{
						backgroundColor: 'rgba(255,255,255,0.7)',
						flex: 1,
						flexDirection: 'column',
						justifyContent: 'center',
						alignItems: 'center'
					}}
				>
					{/* Mina hållplatser logo, custom 'spinner' under logo (brummande buss t.ex) */} 
					<Text style={{ marginBottom: 10, opacity: 1 }}>Mina Hållplatser</Text>
					<Spinner
						size="large"
						color={colors.primary}
						noFlex
					/>
				</View>
			</ImageBackground>
		);
	}

}

export default connect(null, { autoLogin })(SplashScreen);
