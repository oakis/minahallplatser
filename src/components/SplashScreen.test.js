import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import SplashScreen from './SplashScreen';

const mockStore = configureMockStore([thunk]);

global.window.log = () => {};

describe('SplashScreen', () => {
    it('should match snapshot', async () => {
        const wrapper = await shallow(<SplashScreen store={mockStore()} />);
        expect(wrapper.dive()).toMatchSnapshot();
    });
});
