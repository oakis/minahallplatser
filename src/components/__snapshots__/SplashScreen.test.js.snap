// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen should match snapshot 1`] = `
<div>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255,255,255,0.7)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <DefaultFont
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "bold",
          "marginBottom": 10,
          "opacity": 1,
        }
      }
    >
      Mina HÃ¥llplatser
    </DefaultFont>
    <Spinner
      color="#779ECB"
      noFlex={true}
      size="large"
    />
  </View>
</div>
`;
